builds/checkin/api-proxy.yaml

```diff
@@ -3,30 +3,18 @@ pr:
   branches:
     include:
       - master
-jobs:
-
-################################################################################
-  - job: check_run_pipeline
-################################################################################
-    displayName: Check pipeline preconditions (changes ARE in builds or edge-modules/api-proxy-module or mqtt/edgelet-client)
-    pool:
-      vmImage: "ubuntu-16.04"
-    steps:
-      - bash: |
-          git log -m -1 --name-only --first-parent --pretty="" | egrep -i '^(builds|edge-modules/api-proxy-module|mqtt/edgelet-client)'
-          if [[ $? == 0 ]]; then
-            echo "Detected changes inside builds or edge-modules/api-proxy-module or mqtt/edgelet-client folders"
-            echo "##vso[task.setvariable variable=RUN_PIPELINE;isOutput=true]TRUE"
-          fi
-        displayName: Check changes in sources
-        name: check_files
+      - release/*
+  paths:
+    include:
+      - "edge-modules/api-proxy-module/*"
+      - "mqtt/edgelet-client/*"
```

Comment:
this looks weird. `edgelet-client` is a dependecy of `api-proxy-module`. so changes in the `edgelet-client` should not cause checks running for any other crate. we don't run checks when 3rd party dependency changed...

Reply:
I would let @huguesBouvier argue this case :) I just converted the task. 

From my side, I understand what you're saying Denis, but it also makes to have this safeguard against breaking api-proxy. Someone may go and make breaking changes in `edgelet-client`, and fix the broker but forget about api-proxy...

Reply:
I believe if we make changes in edgelet-client we should check api proxy still compiles. Otherwise there is a risk of have a code that doesn't